agimus_controller/tests/test_warm_start_reference.py

```diff
@@ -0,0 +1,64 @@
+import unittest
+
+import numpy as np
+import example_robot_data as robex
+import pinocchio as pin
+
+from agimus_controller.warm_start_reference import WarmStartReference
+from agimus_controller.trajectory import TrajectoryPoint
+
+
+class TestWarmStart(unittest.TestCase):
+    def test_initialization(self):
+        ws = WarmStartReference()
+        self.assertEqual(ws._previous_solution, [])
+
+    def test_generate(self):
+        ws = WarmStartReference()
+        num_points = 10
+        robot = robex.load("ur5")
+        rmodel = robot.model
+        rdata = robot.data
+        ws.setup(rmodel=rmodel)
+
+        initial_q = np.random.randn(rmodel.nq)
+        initial_v = np.random.randn(rmodel.nv)
+        initial_state = TrajectoryPoint(
+            robot_configuration=initial_q, robot_velocity=initial_v
+        )
+
+        random_qs = np.random.randn(num_points, rmodel.nq)
+        random_vs = np.random.randn(num_points, rmodel.nv)
+        random_acs = np.random.randn(num_points, rmodel.nv)
+        reference_trajectory = [
+            TrajectoryPoint(
+                robot_configuration=q, robot_velocity=v, robot_acceleration=a
+            )
+            for q, v, a in zip(random_qs, random_vs, random_acs)
+        ]
+
+        # Create the expected stacked array
+        expected_x0 = np.concatenate([initial_q, initial_v])
+        expected_x_init = np.hstack((random_qs, random_vs))
+        expected_u_init = np.array(
+            [
+                pin.rnea(rmodel, rdata, q, v, a)
+                for q, v, a in zip(random_qs, random_vs, random_acs)
+            ]
+        )
+
+        # Act
+        x0, x_init, u_init = ws.generate(initial_state, reference_trajectory)
+        x0 = np.array(x0)
```

Comment:
`x0` by the definition of the returned values should be `np.array` if the tests are failing without this casting then the class has to be corrected

Reply:
indeed! fixed